graylog2-server/src/main/java/org/graylog/plugins/sidecar/rest/resources/ConfigurationVariableResource.java

```diff
@@ -100,7 +105,8 @@ public Response createConfigurationVariable(@ApiParam(name = "JSON body", requir
                                              @Valid @NotNull ConfigurationVariable request) {
         ValidationResult validationResult = validateConfigurationVariableHelper(request);
         if (validationResult.failed()) {
-            return Response.status(Response.Status.BAD_REQUEST).entity(validationResult).build();
```

Comment:
The former code returned a structured API error like:
```

{
  "errors": {
    "name": [
      "A variable with that name already exists."
    ]
  },
  "failed": true
}
```
And it was identical to the response from the validation.

Now this would return:
```
{
  "type": "ApiError",
  "message": "Cannot update variable. {name=[A variable with that name already exists.]}"
}
```

Reply:
This is based on feedback from @edmundoa, the point was that an Exception makes it more clear that something went wrong. But I can also see your point of a better structured API error. I don't have a strong opinion, we should just have the same type of error under the `/sidecars/*` API namespace.

Reply:
I thought we could pass `validationResults` to the exception to get that into the response body. If that's not possible, we can leave it as it was.

Reply:
Throwing an exception or returning a `Response.status(Response.Status.BAD_REQUEST)` doesn't make a huge difference because the exception mapper will return it into a 400 or 500 anyway.

So I think we can use `Response.status(Response.Status.BAD_REQUEST)` with the proper `validationResults` to get a structured response.